src/tox/session/cmd/run/common.py

```diff
@@ -144,9 +144,15 @@ def report(start: float, runs: List[ToxEnvRunResult], is_colored: bool) -> int:
     def _print(color_: int, message: str) -> None:
         print(f"{color_ if is_colored else ''}{message}{Fore.RESET if is_colored else ''}")
 
-    all_good = True
+    results = []
     for run in runs:
-        all_good &= run.code == Outcome.OK or run.ignore_outcome
+        # a tox run with only skipped envs is considered a failure
+        # that is why we only take successful runs into account
+        if not run.skipped:
+            if run.code == Outcome.OK or run.ignore_outcome:
+                results.append(True)
+        if run.code != Outcome.OK and not run.ignore_outcome:
+            results.append(False)
         duration_individual = [o.elapsed for o in run.outcomes]
```

Comment:
IMHO a better way to express this, and reads better (no longer needed your comment), is to do:

```python
    successful, skipped = [], []
    for run in runs:
        successful.append(run.code == Outcome.OK or run.ignore_outcome) 
        skipped.append(run.skipped)
        duration_individual = [o.elapsed for o in run.outcomes]
     ...

    all_good = all(successful) and not all(skipped)
    if all_good:
       ...
```